### ISSUE TITLE
Reconnect inside socketFor

### ISSUE THREAD

**Issue by evanphx** [thoov/ember-websockets#71]
At present, `socketFor` can return closed Websockets and I think that's a bug. The reason is that https://github.com/thoov/ember-websockets/blob/master/app/services/websockets.js#L60-L67 does not check if the socket in the existing proxy is actually open before returning it.

Adding something like:
```
      if (isWebSocketOpen(existingProxy.socket)) {
        return existingProxy.socket;
      }

      existingProxy.socket.reconnect();

      return existingProxy.socket;
```

When seeing that an `existingProxy` is available would fix the problem.

**Comment by thoov** [comment]
@evanphx Thanks for opening this issue. What you are describing is definitely a bug as `socketFor` should always return an open websocket (pending that it can actually connect to the backend).

I am a little confused without steps to reproduce as the socketFor method works as follows:

Check if there is an existing and open socket: https://github.com/thoov/ember-websockets/blob/master/app/services/websockets.js#L48-L52

If not then create a new socket: https://github.com/thoov/ember-websockets/blob/master/app/services/websockets.js#L58

And if there was an existing proxy (ie there might be previous listeners that we want to keep) we attach this new websocket to the proxy: https://github.com/thoov/ember-websockets/blob/master/app/services/websockets.js#L60-L67 or if there was no existing proxy we create a new one: https://github.com/thoov/ember-websockets/blob/master/app/services/websockets.js#L69-L7.

So from what you are describing is the way that it should be working. Are you able to reproduce what you are seeing?

**Comment by evanphx** [comment]
Ah! I probably missed that new code because I've had this issue with older versions of ember-websockets. I'll have to try again with a version that has this code in it. Right now, I'm using a custom vendored version that looks like:

```javascript

    let existingProxy = this.get(`sockets.${cleanedUrl}`);

    if (existingProxy) {
      if (isWebSocketOpen(existingProxy.socket)) {
        return existingProxy.socket;
      }

      existingProxy.socket.reconnect();

      return existingProxy.socket;
    }

...
}
```

**Comment by thoov** [comment]
@evanphx I see. Yes, this issue was brought up with #68 and I just recently fix it. The fix should be in 3.0.0
